### ISSUE TITLE
Toot compose view jumps around while typing in text field on iPad Pro

### ISSUE THREAD

**Issue by cortices** [DagAgren/toot#411]
When typing a new toot on iPad Pro, the toot composing scrollview moves scroll position to position the cursor at the very top of the screen. But while typing, every second character/change alternates between that position and one a few pixels up from it. This behaviour exhibits as the view jittering up and down relative to typing speed.

The positioning of the cursor at the top of the view, not just “in view”, also means that when typing any toot longer than 1 line, it is impossible to keep higher lines in view while typing, as the scroll position jumps as soon as anything is typed.

Proposed fix would be to ensure the cursor in the text field is within the viewing area (not out of scroll view and not hidden by keyboard), but not enforce positioning further than that.

Video of behaviour will be sent via mastodon.

**Comment by DagAgren** [comment]
Toot does no manipulation at all of the scroll position on the posting screen, so this seems to be caused by something else. Not sure what yet though.

**Comment by DagAgren** [comment]
What is the exact device, iOS version and Toot! version?

**Comment by cortices** [comment]
This is visible on iPad Pro 12.9” (3rd generation), (“iPad8,5”). It has been the same since I first started using toot on the device, maybe 6-8 months ago? Currently on v1.13. I’m using a regular Apple keyboard on Australian English. It has the regular grey bar of shortcuts and typing suggestions above the keys as shown in video.
Something that many other iPad apps do is when the keyboard is visible, the modal window (of the toot composer) shifts upwards (Or expands vertically), so as to make use of the more limited vertical space. Toot leaves the ~25% margin above the toot composer window even when typing, meaning there’s a severely limited usable space for composing the toot. Seeing as this behaviour isn’t directly caused by manipulating scroll position, I suspect moving the toot composer window to provide more space might possibly fix it? Or at least provide another datum while improving usability.

**Comment by cortices** [comment]
Uh, just ignore what I said, because it does do that shift upwards?? I think it’s system behaviour? But it only did it after I made some very particular swipe or tap on some part of the modal by accident???? And it still moves the cursor to the top of the view aggressively. Are you sure there’s no code forcing the text field cursor to be visible? Really odd

**Comment by DagAgren** [comment]
Oh, what I do do is resize the text field to make it larger as you type. That seems to be interfering with the iPad's attempts to keep the tex visible, I think?

**Comment by DagAgren** [comment]
Ok, this was weird.

It seems I changed the presentation style of that view from the default "pageSheet" to "formSheet", because on the smaller iPads pageSheet made it a bit gigantic, and formSheet had a slightly smaller and nicer size.

However, it seems formSheet in iOS 13 is completely broken, and is causing all the mad scrolling and positioning problems you describe. I got the same ones on the simulator too.

I changed it back to pageSheet and will just try to live with it being a bit big on smaller iPads, I guess. It seems to fix all the problems.
